### DISCUSSION TITLE
Some small questions about the algorithms.

### DISCUSSION THREAD

**Discussion by nesou2** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#244] [General]
Hey there! Excellent algorithm, I have started using stable-diffusion recently and I was hoping that someone would have implemented something like this. The tiled diffusion seems to have a great mix of local details, whilst also maintain global consistency. I have a few questions to ask!

Firstly, why does using region prompt control disable tiled diffusion?
As I understand it, when you do region prompt control you will make a box which will be passed through the diffusion model, with the specific prompt. Why not instead do tile diffusion, and apply the prompt to the tiles in the specified region? 

Secondly, I read the papers for multidiffuison and mixture of diffusers and they seem to suggest essentially the same method, justified in different ways. If I understand correctly, both methods will have several diffusion models that take different segments of the image as input. They, you take some fixed, linear (with each pixel weighing the models differently) combination of the models. Have I understood correctly? If so, how does the implementation differ?

Finally, I was just curious why you didn't implement custom mask shapes, instead opting for the rectangle mask (if I understood correctly). Is there a reason for this? 

Again, thanks for the great extension!

**Comment by pkuliyi2015** [comment]
Thanks for your attention.

1) The region prompt control disables the auto tiling by default, **but you can enable it by checking the "draw full canvas background"**. As many users just want to do compositional generation, disabling the auto tiling by default can save their time, as the auto tiling produces many unnecessary tiles that can be covered by their custom regions.

2) The key difference of the two paper is that multidiffusion is averaging the **denoised latents** with a sum of torch.ones as weights, while mixture of diffusers is averaging the UNet **predicted noises** with a sum of gaussian-based weights. Note that there are still many steps to utilize predicted noises to produce denoised latents. I'm not sure if any of them is non-linear. If all of these steps are linear, the two can be mathematically identical. 

But from the engineering perspective, when do a simple hijack into the webui pipeline, the multidiffusion method allows you to preview a small region in t2i (as it produces 0s for all uncovered areas), and the mixture of diffusers method allows you to redraw a small region in i2i and leaves others unchanged. So the two methods are still different. It is very complicated to use one algorithm to achieve the above 2 purposes.

3) UNet only accepts rectangular tensors. While custom mask shape can be implemented as a post-processing after each UNet pass, but it is very inaccurate as the masks will randomly affect each other. I don't think the flexible mask shapes will be as powerful as ControlNet (e.g., segmentation control). As Tiled Diffusion supports ControlNet, I have no need to implement custom mask shapes.
